Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/opensaml/extraction/OpenSAMLAssertionExtractorImpl.java

```diff
@@ -252,17 +254,26 @@ private void compareAttribute(Attribute attribute, AssertionType target, Attribu
         } else if (attribute.getName().equals(SamlConstants.ACCESS_CONSENT_ATTR)) {
             List<String> accessConsentId = transformXMLtoString(attribute.getAttributeValues());
             target.getSamlAuthzDecisionStatement().getEvidence().getAssertion().getAccessConsentPolicy()
-            .addAll(accessConsentId);
+                    .addAll(accessConsentId);
             LOG.debug("Assertion.SamlAuthzDecisionStatement.Evidence.Assertion.AccessConsentPolicy = {}",
-                accessConsentId);
+                    accessConsentId);
 
         } else if (attribute.getName().equals(SamlConstants.INST_ACCESS_CONSENT_ATTR)) {
             List<String> instAccessConsentId = transformXMLtoString(attribute.getAttributeValues());
             target.getSamlAuthzDecisionStatement().getEvidence().getAssertion().getInstanceAccessConsentPolicy()
-            .addAll(instAccessConsentId);
+                    .addAll(instAccessConsentId);
             LOG.debug("Assertion.SamlAuthzDecisionStatement.Evidence.Assertion.InstanceAccessConsentPolicy = {}",
-                instAccessConsentId);
-
+                    instAccessConsentId);
+        } else if (attribute.getName().equals(SamlConstants.ATTRIBUTE_NAME_XUA_ACP)) {
+            String acpValue = getAttributeValue(attribute);
+            if(NullChecker.isNotNullish(acpValue)) {
+                target.setAcpAttribute(acpValue);
+            }
+        } else if (attribute.getName().equals(SamlConstants.ATTRIBUTE_NAME_XUA_IACP)) {
+            String iacpValue = getAttributeValue(attribute);
+            if(NullChecker.isNotNullish(iacpValue)) {
```

Comment:
Use stringUtil.IsNotEmpty to check trailing/leading space along with null.  We can update NullChecker method to include that method instead.